components/feature/search/src/main/java/mozilla/components/feature/search/SearchUseCases.kt

```diff
@@ -20,15 +24,19 @@ import mozilla.components.browser.state.store.BrowserStore
  * this (optional) lambda will be invoked to create a [Session]. The default implementation creates a [Session] and adds
  * it to the [SessionManager].
  */
+@Suppress("LongParameterList")
 class SearchUseCases(
     context: Context,
     store: BrowserStore,
     searchEngineManager: SearchEngineManager,
     sessionManager: SessionManager,
+    private val bookmarksStorage: BookmarksStorage,
```

Comment:
I'm not sure if this should be mandatory as not all browsers will have `Bookmarks`, but all will do search for example [Reference browser](https://github.com/mozilla-mobile/reference-browser) doesn't have bookmarks but it need search